doc/REFERENCE.md

```diff
@@ -259,81 +198,157 @@ This option adds additional attributes to each vector feature, the attributes ar
-and is not recommended. Most _OpenSpaceNet_ models are designed to work at a certain resolution and do not benefit from pyramiding.`
+This option will enable using pyramids in feature detection. This means that the 
+image will be repeatedly resampled down by a factor of 2 with sliding window 
+detection run on each reduced image. This option will result in much longer run 
+time and is not recommended. Most _OpenSpaceNet_ models are designed to work at 
+a certain resolution and do not benefit from pyramiding.`
+
+When this is given, `--window-size` and `--window-step` arguments may be 
+specified, but only the first option for each will be used.
+
+<a name="segmentation" />
+
+### Segmentation Options
+
+Segmentation options contain options for raster-to-polygon conversion for
+segmentation models. These are not mandatory and are only applicable to
+segmentation models. If set for a different model type, a warning will be 
+displayed.
+
+##### --r2p-method METHOD
+This is the raster-to-polygon approximation method.
+Valid values are: `none`, `simple`, `tc89-l1`, `tc89-kcos`. The default value
+is `simple`.
+
+The methods are as follows:
+
+* `none`: Stores absolutely all the contour points. That is, any 2 subsequent 
+points _(x<sub>1</sub>,y<sub>1</sub>)_ and _(x<sub>2</sub>,y<sub>2</sub>)_ of 
+the contour will be either horizontal, vertical or diagonal neighbors, that is, 
+<em>max(|x<sub>1</sub> - x<sub>2</sub>|, |y<sub>2</sub>-y<sub>1</sub>|) = 1</em>.
+
+* `simple`: Compresses horizontal, vertical, and diagonal segments and leaves 
+only their end points. For example, an up-right rectangular contour is encoded 
+with 4 points.
+
+* `tc89-l1`:  Applies the <em>1 curvature</em> Teh-Chin chain approximation 
+algorithm.
+
+* `tc89-kcos`: Applies the <em>k cosine</em> Teh-Chin chain approximation 
+algorithm.
+
 
-When this is given, `--window-size` and `--window-step` arguments may be specified, but only the first option for each will be used.
+The Teh-Chin approximations are as described in:
 
-##### Size parameters
+> Teh, C-H and Chin, Roland T. <b>On the detection of dominant points on digital curves.</b> <br/> 
+> IEEE Transactions on Pattern Analysis and Machine Intelligence, 1989, vol. 11, num. 8, pp. 859-872 
 
-Some models may require that the imagery be resized or padded to fit the target model.  When performing these actions, use `--resampled-size` and `--window-size` to add padding and resizing, respectively.  See the image below for guidance.
+##### --r2p-accuracy
 
-![Resizing parameters](modelsize.svg)
+This argument specifies the polygon approximation accuracy. This is the maximum 
+distance between the original curve and its approximation. The default value is 3.
+
+
+##### --r2p-min-area
+
+This argument sets the minimum area of a polygon in pixels. The default value
+is 0.
 
 <a name="filter" />
 
-### Filter Options
+### Filtering Options
```

Comment:
To `--include-region` / `--exclude-region` / `--region`, consider adding language to be specific that we do not filter at the  detect level.  E.g.
>_Note:_ Filtering is an optimization that is not performed at the detection level.  This is will impact detection models, such as DetectNet, where one window may contain result in many detections.  This may cause detects far from the include/exclude boundary to be included in the output.  Post processing will be required to remove these detections.

Reply:
Good call. This is the place to explain what region filtering is for.